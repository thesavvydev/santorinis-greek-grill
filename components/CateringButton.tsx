import { useRouter } from "expo-router";
import { Text } from "react-native";
import DoubleButton from "./DoubleButton";

enum CATERING_PACKAGES {
  "bowl-buffet",
  "pita-buffet",
  "boxed-meals",
  "ultimate-combo",
}

type CateringButtonType = {
  cateringPackage: keyof typeof CATERING_PACKAGES;
};

export default function CateringButton({
  cateringPackage,
}: CateringButtonType) {
  const router = useRouter();

  return (
    <DoubleButton
      onPress={() => {
        router.push(
          `/(tabs)/catering/catering-form-modal?cateringPackage=${cateringPackage}`
        );
      }}
    >
      <Text className="text-white font-bold uppercase">Request Catering</Text>
    </DoubleButton>
  );
}
