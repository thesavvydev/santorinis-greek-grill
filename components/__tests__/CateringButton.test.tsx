import * as React from "react";
import { fireEvent, render, screen } from "@testing-library/react-native";
import CateringButton from "../CateringButton";
import * as Router from "expo-router";

describe("CateringButton", () => {
  it(`matches snapshot`, () => {
    const tree = render(
      <CateringButton cateringPackage="boxed-meals" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders text`, () => {
    render(<CateringButton cateringPackage="boxed-meals" />).toJSON();

    screen.getByText("Request Catering");
  });

  it("on press routes to catering-form-modal", () => {
    const push = jest.fn();
    jest.spyOn(Router, "useRouter").mockReturnValue({
      push,
      back: function (): void {
        throw new Error("Function not implemented.");
      },
      canGoBack: function (): boolean {
        throw new Error("Function not implemented.");
      },
      navigate: function <T extends string | object>(
        href: Router.Href<T>
      ): void {
        throw new Error("Function not implemented.");
      },
      replace: function <T extends string | object>(
        href: Router.Href<T>
      ): void {
        throw new Error("Function not implemented.");
      },
      dismiss: function (count?: number): void {
        throw new Error("Function not implemented.");
      },
      dismissAll: function (): void {
        throw new Error("Function not implemented.");
      },
      canDismiss: function (): boolean {
        throw new Error("Function not implemented.");
      },
      setParams: function <T extends Router.Routes>(
        params: Partial<Router.RouteParamInput<T>>
      ): void {
        throw new Error("Function not implemented.");
      },
    });

    render(<CateringButton cateringPackage="boxed-meals" />);

    fireEvent.press(screen.getByText("Request Catering"));

    expect(push).toHaveBeenCalledWith(
      "/(tabs)/catering/catering-form-modal?cateringPackage=boxed-meals"
    );
  });
});
