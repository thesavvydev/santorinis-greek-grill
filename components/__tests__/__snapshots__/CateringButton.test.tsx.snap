// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CateringButton matches snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#b45309",
      "opacity": 1,
      "paddingBottom": 16,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 16,
      "shadowOffset": {
        "height": 5,
        "width": 5,
      },
      "shadowOpacity": 10,
      "shadowRadius": 0,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
        },
        {
          "fontWeight": "700",
        },
        {
          "textTransform": "uppercase",
        },
      ]
    }
  >
    Request Catering
  </Text>
</View>
`;
